screencaptor/src/main/java/com/wealthfront/screencaptor/views/modifier/ViewDataProcessor.kt

```diff
@@ -0,0 +1,52 @@
+package com.wealthfront.screencaptor.views.modifier
+
+import android.graphics.drawable.Drawable
+import android.view.View
+import android.widget.ImageView
+import android.widget.TextView
+
+internal class DefaultViewDataProcessor : ViewDataProcessor {
+
+  override fun modifyViews(view: View, viewDataModifiers: Set<DataModifier>): Set<DataModifier> {
+    val listOfInitialStateModifier = mutableSetOf<DataModifier>()
+    viewDataModifiers.forEach { modifier ->
+      val initialStateModifier: DataModifier
+      when (modifier) {
+        is TextViewDataModifier -> {
+          val textView = view.findViewById<TextView>(modifier.id)
+          initialStateModifier = TextViewDataModifier(modifier.id, textView.text)
+          textView.text = modifier.data
+          listOfInitialStateModifier.add(initialStateModifier)
+        }
+        is ImageViewDataModifier -> {
+          val imageView = view.findViewById<ImageView>(modifier.id)
+          initialStateModifier = ImageViewDataModifier(modifier.id, imageView.drawable)
+          imageView.setImageDrawable(modifier.data)
+          listOfInitialStateModifier.add(initialStateModifier)
+        }
+      }
```

Comment:
I actually had this as a sealed class so that its exhaustive. But since I want us to be able to provide a custom processor with custom data modifiers (custom implementation out of the sealed class), we can't really be exhaustive with the modifiers. One of the tradeoffs of extensibility here.

Reply:
Will add a throw statement here.